From 2dfc9bead4edabcf7eeef876e56e0b4e123fdb18 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:50:08 +0800
Subject: [PATCH 61/62] Revert "drm/komeda: return early if
 drm_universal_plane_init() fails."

This reverts commit c977d63b8cc45a1ca4ce438c072af43af6f4aa6b.
---
 drivers/gpu/drm/arm/display/komeda/komeda_plane.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
index a5f57b38d193..98e915e325dd 100644
--- a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
+++ b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
@@ -274,10 +274,8 @@ static int komeda_plane_add(struct komeda_kms_dev *kms,
 
 	komeda_put_fourcc_list(formats);
 
-	if (err) {
-		kfree(kplane);
-		return err;
-	}
+	if (err)
+		goto cleanup;
 
 	drm_plane_helper_add(plane, &komeda_plane_helper_funcs);
 
-- 
2.27.0

